server/src/main/java/io/crate/role/TransportDropRoleAction.java

```diff
@@ -158,6 +161,21 @@ static boolean dropRole(Metadata.Builder mdBuilder, String roleNameToDrop) {
         return role != null;
     }
 
+    @VisibleForTesting
+    static void ensureUserDoesNotOwnForeignServers(Metadata metadata, String roleName) {
```

Comment:
Could we move this into the task and then test that, to avoid visibility changes for tests?